### PR TITLE
Move Iceberg failure recovery test to separate job

### PR DESCRIPTION
**Pull request by losipiuk**
## Description

Move Iceberg failure recovery test to separate job

> Is this change a fix, improvement, new feature, refactoring, or other?

test refactoring

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

test code, iceberg connector

> How would you describe this change to a non-technical end user or system administrator?

N/A 

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

### CODE REVIEW

File: plugin/trino-iceberg/pom.xml [hunk 1]
```diff
@@ -301,34 +301,70 @@
...
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <excludes>
+                                <exclude>**/TesIceberg*FailureRecoveryTest.java</exclude>
+                            </excludes>
+                        </configuration>
+                    </plugin>
+
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
```
Comment (aczajkowski) [inline plugin/trino-iceberg/pom.xml:323]: Shouldn't we leave them in plugin management section ?

**Review by aczajkowski — APPROVED** [review]
LGTM % one question

File: plugin/trino-iceberg/pom.xml (hunk 1 above)
Comment (losipiuk) [reply plugin/trino-iceberg/pom.xml:323]: IDK. I copied approach form pom.xml for Hive connector.

Comment (losipiuk) [reply plugin/trino-iceberg/pom.xml:323]: Damn - it looks like excluded tests still executed for default profile ...

Comment (losipiuk) [reply plugin/trino-iceberg/pom.xml:323]: typo

**Review by arhimondr — APPROVED** [review]

File: plugin/trino-iceberg/pom.xml [hunk 2]
```diff
@@ -301,34 +301,70 @@
...
-        <pluginManagement>
-            <plugins>
-                <plugin>
-                    <groupId>org.apache.maven.plugins</groupId>
-                    <artifactId>maven-enforcer-plugin</artifactId>
-                    <configuration>
-                        <rules>
-                            <requireUpperBoundDeps>
-                                <excludes combine.children="append">
-                                    <exclude>com.google.guava:guava</exclude>
```
Comment (findepi) [inline plugin/trino-iceberg/pom.xml:314]: Why this appears as a change here?
did you mean a prep commit?

**Review by findepi — APPROVED** [review]

File: plugin/trino-iceberg/pom.xml (hunk 2 above)
Comment (losipiuk) [reply plugin/trino-iceberg/pom.xml:314]: It is verbatim move from <build> -><pluginManagement> to default profile

Comment (losipiuk) [reply plugin/trino-iceberg/pom.xml:314]: I extracted separate commit. But it still shows in diff as it is different level of indentation.

**Pull request merged**